### PR TITLE
[feat] Deepbook-sui: add fees and revenue adapter for deepbook v3

### PR DESCRIPTION
**Pull request by Shaileshkhote**
## Notes
- Consumes 1.06 Dune Credits per run 

## Summary
- Adds a Sui fees adapter for DeepBook under `fees/deepbook-sui`.
- Tracks gross trading fees, DEEP burn revenue/holder revenue, and supply-side payouts from maker rebates and referral rewards.

## Changes
- Added a Dune-backed `fees/deepbook-sui` adapter for `CHAIN.SUI`.
- Uses DeepBook V3 `OrderFilled` events for maker/taker trading fees.
- Uses `DeepBurned` events for revenue and holders revenue.
- Uses `ReferralFeeEvent` and `RebateEventV2` for supply-side revenue.
- Prices DEEP from the DeepBook `DEEP/USDC` pool and SUI from the `SUI/USDC` pool; base-token rebate/referral amounts are converted through same-day DeepBook pool fill prices.

## Methodology
- `dailyFees`: all maker and taker fees paid by users on DeepBook V3 trades.
- `dailyRevenue`: DEEP tokens burned by DeepBook from accumulated trading fees.
- `dailyHoldersRevenue`: same as revenue, because burned DEEP reduces token supply.
- `dailyProtocolRevenue`: `0`, because no direct treasury revenue is counted.
- `dailySupplySideRevenue`: maker rebates claimed plus referral rewards.
- Burns, rebates, and referral rewards are counted on their event dates, so they may not reconcile exactly with same-day trading fees.

### CODE REVIEW

File: fees/deepbook-v3/index.ts [hunk 1]
```diff
@@ -0,0 +1,263 @@
...
+prices AS (
+  SELECT
+    date,
+    SUM(CASE WHEN pool_id = '${DEEP_USDC_POOL}' THEN quote_qty / 1e6 END)
+      / NULLIF(SUM(CASE WHEN pool_id = '${DEEP_USDC_POOL}' THEN base_qty / 1e6 END), 0) AS deep_usd,
+    SUM(CASE WHEN pool_id = '${SUI_USDC_POOL}' THEN quote_qty / 1e6 END)
+      / NULLIF(SUM(CASE WHEN pool_id = '${SUI_USDC_POOL}' THEN base_qty / 1e9 END), 0) AS sui_usd
+  FROM order_fills
+  GROUP BY 1
+),
```
[the remark this answers is not in the document]
Comment (noateden) [reply fees/deepbook-v3/index.ts:116]: @Shaileshkhote can you check this claim?

[the remark this answers is not in the document]
Comment (Shaileshkhote) [reply fees/deepbook-v3/index.ts:116]: > @Shaileshkhote can you check this claim?

The claim is right, though it wont happen as its edge case adding a fix for it

**Comment by Shaileshkhote** [comment]
@noateden I have addresses all the coderabbit reviews and fixed them,
please check and test once

File: fees/deepbook-v3/index.ts [hunk 2]
```diff
@@ -0,0 +1,263 @@
...
+        ELSE 0
+      END
+      +
+      CASE
+        WHEN pp.qusd IS NULL THEN 0
+        WHEN NOT f.maker_fee_is_deep AND NOT f.taker_is_bid THEN f.maker_fee / POWER(10, pp.qdec) * pp.qusd
+        WHEN NOT f.maker_fee_is_deep AND f.taker_is_bid AND f.base_qty > 0
+          THEN f.maker_fee * f.quote_qty / NULLIF(f.base_qty, 0) / POWER(10, pp.qdec) * pp.qusd
+        ELSE 0
+      END
```
[the remark this answers is not in the document]
Comment (Shaileshkhote) [reply fees/deepbook-v3/index.ts:161]: addressed and fixed

File: fees/deepbook-v3/index.ts [hunk 3]
```diff
@@ -0,0 +1,263 @@
...
+        se.deep_fee / 1e6 * pp.deep_usd
+        + CASE WHEN pp.qusd IS NULL THEN 0 ELSE se.quote_fee / POWER(10, pp.qdec) * pp.qusd END
+        + CASE WHEN pp.qusd IS NULL OR pp.quote_per_base_raw IS NULL THEN 0 ELSE se.base_fee * pp.quote_per_base_raw / POWER(10, pp.qdec) * pp.qusd END
+      ELSE 0 END
+    ) AS rebates_claimed_usd,
+    SUM(CASE WHEN se.kind = 'referral' AND pp.quote_kind = 'UNKNOWN' THEN 1 ELSE 0 END) AS unknown_referral_quote_events,
+    SUM(CASE WHEN se.kind = 'rebate' AND pp.quote_kind = 'UNKNOWN' THEN 1 ELSE 0 END) AS unknown_rebate_quote_events
+  FROM supply_events se
+  JOIN priced_pools pp ON se.date = pp.date AND se.pool_id = pp.pool_id
+  GROUP BY 1
```
[the remark this answers is not in the document]
Comment (Shaileshkhote) [reply fees/deepbook-v3/index.ts:187]: addressed and fixed

File: fees/deepbook-v3/index.ts [hunk 4]
```diff
@@ -0,0 +1,263 @@
...
+)
+
+SELECT
+  SUM(tf.daily_fees_usd) AS daily_fees_usd,
+  SUM(COALESCE(b.deep_burned * pr.deep_usd, 0)) AS daily_revenue_usd,
+  SUM(COALESCE(ss.referral_fees_usd, 0)) AS referral_fees_usd,
+  SUM(COALESCE(ss.rebates_claimed_usd, 0)) AS rebates_claimed_usd,
+  SUM(tf.unknown_quote_pair_fee_fills) AS unknown_quote_pair_fee_fills,
+  SUM(COALESCE(ss.unknown_referral_quote_events, 0)) AS unknown_referral_quote_events,
+  SUM(COALESCE(ss.unknown_rebate_quote_events, 0)) AS unknown_rebate_quote_events
```
[the remark this answers is not in the document]
Comment (Shaileshkhote) [reply fees/deepbook-v3/index.ts]: addressed and fixed

**Review by noateden — APPROVED** [review]
